src/Core/src/Response.php

```diff
@@ -85,24 +86,83 @@ public function resolve(?float $timeout = null): bool
                 }
             }
 
-            $statusCode = $this->httpResponse->getStatusCode();
+            return $this->handleStatus();
         } catch (TransportExceptionInterface $e) {
             throw $this->resolveResult = new NetworkException('Could not contact remote server.', 0, $e);
         }
+    }
 
-        if (500 <= $statusCode) {
-            throw $this->resolveResult = new ServerException($this->httpResponse);
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $responses
+     * @param float|null $timeout   Duration in seconds before aborting. When null wait
+     *                              until the end of execution. Using 0 means non-blocking
+     *
+     * @return iterable<self, bool> whether the request is executed or not
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function resolveAll(iterable $responses, float $timeout = null): iterable
+    {
+        /** @var self[] $responseMap */
+        $responseMap = [];
+        $httpResponses = [];
+        $httpClient = null;
+        foreach ($responses as $response) {
+            if (null !== $response->resolveResult) {
+                if (\is_bool($response->resolveResult)) {
+                    yield $response => $response->resolveResult;
+                } else {
+                    yield $response => true;
```

Comment:
I'm not happy with this. it hides exceptions and user may think (wrongly) the response is successfully proceeded.

I can't return false neither, user may think that response is not yet proceeded and call this method again and again...

returning the exception is not really usable.

I think I'll rename this method into `multiplex`, yielding only the response. and let user calling `resolve` (or fetching a getter) on each item

Reply:
fixed